pkg/operator/ceph/object/cosi/spec.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cosi
+
+import (
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	defaultCOSISideCarImage    = "gcr.io/k8s-staging-sig-storage/objectstorage-sidecar/objectstorage-sidecar:v20230130-v0.1.0-24-gc0cf995"
+	defaultCephCOSIDriverImage = "quay.io/ceph/cosi:v0.1-alpha"
+)
+
+func createCephCOSIDriverDeployment(cephCOSIDriver *cephv1.CephCOSIDriver) (*appsv1.Deployment, error) {
+	cosiPodSpec, err := createCOSIPodSpec(cephCOSIDriver)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create cosi pod spec")
+	}
+	strategy := appsv1.DeploymentStrategy{
+		Type: appsv1.RecreateDeploymentStrategyType,
+	}
+
+	replica := int32(1)
+	minReadySeconds := int32(30)
+	progressDeadlineSeconds := int32(600)
+	revisionHistoryLimit := int32(3)
+
+	cephcosidriverDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cephCOSIDriver.Name,
```

Comment:
We only allow a single COSI driver, right? We could simplify restricting for multiple COSI drivers if we override the name with a const name. But we can consider in a future PR.